backend/layers/processing/rollback.py

```diff
@@ -0,0 +1,301 @@
+                logger.info(f"Failed to rollback DatasetVersion {dataset_version_id}, {e}")
+                continue
+            rolled_back_datasets.append(rolled_back_dataset)
+        self._clean_up_rolled_back_datasets(rolled_back_datasets)
+
+    def rollback_private_dataset(
+        self, dataset_version_id: DatasetVersionId, collection_version_id: CollectionVersionId = None
+    ) -> DatasetVersion:
+        """
+        For a given DatasetVersionId and unpublished CollectionVersionId, rolls back the associated Dataset in the
+        CollectionVersion to its previous, most recently created DatasetVersion and deletes the given DatasetVersion.
+
+        :param dataset_version_id: DatasetVersionId of the DatasetVersion to rollback
+        :param collection_version_id: CollectionVersionId of the CollectionVersion to rollback the DatasetVersion
+        in. If not passed in, the CollectionVersionId will be determined from the DatasetVersionId.
+        :return: DatasetVersion that was rolled back
+        """
+        cv_id = collection_version_id
+        dataset_version = self.business_logic.get_dataset_version(dataset_version_id)
+        if cv_id is None:
+            # account for collection potentially having a migration revision and a non-migration revision
+            collection_versions = self.business_logic.get_unpublished_collection_versions_from_canonical(
+                dataset_version.collection_id
+            )
+            for cv in collection_versions:
+                if cv_id is not None:
+                    break
+                for dataset in cv.datasets:
+                    if dataset.version_id == dataset_version_id:
+                        cv_id = cv.version_id
+                        break
+        if cv_id is None:
+            raise CollectionNotFoundException(
+                f"An Associated CollectionVersion not found for DatasetVersion {dataset_version_id}"
+            )
+        self.business_logic.restore_previous_dataset_version(cv_id, dataset_version.dataset_id)
+        return dataset_version
+
+    def rollback_public_collections(self) -> None:
+        """
+        Rollback each public collection to its previous, most recently published CollectionVersion (if one exists).
+        This will restore the state of public collections to their pre-migration state. Then, triggers deletion of the DB
+        references and S3 assets for the rolled back CollectionVersions and their associated DatasetVersions.
+        """
+        filter = CollectionQueryFilter(is_published=True)
+        collections = self.business_logic.get_collections(filter)
+        rolled_back_collection_versions = []
+        for collection in collections:
+            try:
+                rolled_back_collection_version = self.rollback_public_collection(collection.collection_id)
+            except Exception as e:
+                logger.info(f"Failed to rollback Collection {collection.collection_id}, {e}")
```

Comment:
- change `info` to `exception` and remove `{e}`. 
- add that it was a public collection to error message